backend/tests/live/test_live_api_sequence.py

```
@@ -0,0 +1,46 @@
+import time
+import urllib.error
+import urllib.request
+from pathlib import Path
+from typing import Any
+
+
+def read_json(url: str, *, attempts: int = 12) -> dict[str, Any]:
+    last_error: Exception | None = None
+    for _ in range(attempts):
+        try:
+            with urllib.request.urlopen(url, timeout=5) as response:
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)